lib/MusicBrainz/Server/Entity/PartialDate.pm

```diff
@@ -1,6 +1,8 @@
 package MusicBrainz::Server::Entity::PartialDate;
 use Moose;
 
+extends 'MusicBrainz::Server::Entity';
```

Comment:
I don't think it's correct to extend `MusicBrainz::Server::Entity`, since it's not exactly an entity in the way that's used, and has other effects like adding `entityType` to the JSON.

I think we should just define a custom `TO_JSON` method in this file that serializes the year/month/day.

But I'm still wondering how to trigger this error. :)

Reply:
How to reproduce: Checkout the commit preceding 23e11bdd65d9d5c92328af47496bb256daaa8255, compile resources (against outdated scripts), checkout master, run the server, try to edit any entity with either date period or partial date set.

This bug cannot be triggered on production/beta/master with up-to-date compilation of resources.

Reported in https://tickets.metabrainz.org/browse/MBS-9446

Other solution with custom `TO_JSON` method proposed in PR #541.